portfolio/src/main/webapp/script.js

```diff
@@ -45,9 +50,216 @@ function getServerMessage() {
 }
 
 /** Creates an <li> element containing text. */
-function createListComments(text){
+function createListComments(commentJson){
     const liElement = document.createElement('li');
-    liElement.innerText = text.comment;
+    liElement.innerText = commentJson.comment;
+
+    const deleteButton = document.createElement('button');
+    deleteButton.innerText = 'Delete';
+    deleteButton.addEventListener('click',() => {
+        deleteComment(commentJson);
+        liElement.remove();
+    });
 
+    liElement.appendChild(deleteButton);
     return liElement;
 }
+
+function deleteComment(commentJson){
+    const params = new URLSearchParams();
+    params.append('id',commentJson.id);
+    fetch('/delete-data', {method: 'POST', body: params});
+}
+
+/** Creates a map and adds it to the page. */
+function createMap(){
```

Comment:
Move the map code to a separate JS file than the comments code. They're doing separate things and there's less chance of errors (e.g., accidentally using the same variable) if they're in separate file scopes.